### DISCUSSION TITLE
Create sites hierarchically nested

### DISCUSSION THREAD

**Discussion by emka-nh** [netbox-community/netbox#18121] [Ideas]
The limitation of prefixes (and VLANs) to only one single site leads to some mind fuzzing workarounds (#10854) and is not solved satisfactorily (#8619). According to documentation a site represents a building, but I cannot create VLANs, VRFs and VLAN groups for every building on my campus. Even worse: If I do, I don't see the available prefixes in site-view, wich would be a great info to get there.

How about redesigning sites/site groups to an approach like regions and locations, where they can be hierarchically nested? Would this be possible by just giving sites a "parent" link like regions and locations?

**Comment by candlerb** [comment]
> According to documentation a site represents a building

Not necessarily. A entire campus can be a single Site, and you can use Locations for buildings and further subdivisions such as rooms, floors etc.

https://netboxlabs.com/docs/netbox/en/stable/models/dcim/site/

> "How you choose to employ sites when modeling your network may vary depending on the nature of your organization, but generally a site will equate to a building **or campus.**" [my emphasis]

**Reply by emka-nh** [reply]
I thought about that, but as @empusas mentioned in #8619 a subnet (aka prefix) can span over cities and regions. In my case we have a MPLS connecting two locations in different cities on layer 2.

Creating a pseudo site doesn't give me the possibility to have sites with addresses and GPS coordinates resembling my reality and sharing one subnet.

In my thinking it would be also more straightforward. Compare https://netboxlabs.com/docs/netbox/en/stable/features/facilities/ to mine:

![image](https://github.com/user-attachments/assets/2a6f0e18-65f3-41eb-9545-2f7f4e59e1f6)
old

![image](https://github.com/user-attachments/assets/2d3199c8-1d11-4517-b481-17188489894e)
new
